src/lambdas/archive-user-logs/src/utils/uploadToS3.ts

```diff
@@ -0,0 +1,20 @@
+import { S3Client, PutObjectCommand } from "@aws-sdk/client-s3"
+import type { Event } from "../types"
+
+const generateObjectPath = (data: Event): string => {
+  const date = new Date()
+  const formatedDate = date.toLocaleDateString("en-GB").split("/").reverse().join("") // '20211124'
+
+  return `${data.logGroup}/${formatedDate}/${data.logStream}/${data.logEvents[0].timestamp}`
+}
+
+export const uploadToS3 = async (data: Event, bucket: string, region: string): Promise<void> => {
+  const client = new S3Client({ region })
```

Comment:
It would be nice to reuse `AwsS3Gateway` from `shared` package here. 

```
const objectKey = generateObjectPath(data)
await new AwsS3Gateway(config).forBucket(bucket).upload(objectKey, JSON.stringify(data))
```

Reply:
This actually needs more changes. S3 client or AwsS3Gateway should be created in `index.ts` file and passed to this function.
An example of this is in `src/lambdas/retrieve-event-from-s3/src/index.ts`